File: 2021/Day2/part2.rb
#!/usr/bin/env ruby

input = File.open('input').readlines.map(&:strip)

depth = 0
horizontal = 0
aim = 0

input.each do |ins|
  direction, distance = ins.split
  case direction
  when "forward"
    horizontal += distance.to_i
    depth += aim * distance.to_i
  when "up"
    aim -= distance.to_i
  when "down"
    aim += distance.to_i
  end
end

p depth * horizontal

File: 2024/day1/part2.py
#!/usr/bin/env python3

with open('input', 'r') as file:
    lines = [line.strip() for line in file.readlines()]

a = set()
b = []

for line in lines:
    i, j = line.split()
    a.add(int(i))
    b.append(int(j))

score = 0

for n in b:
    if n in a:
        score += n

print(score)
